Feat: StatusTag 마크업 추가

wherehows opened the pull request:
## 이슈 번호 <!-- 관련된 이슈의 번호를 # 뒤에 입력해주세요  -->

- close: #3 

## 특이 사항

### 컴포넌트 추상화 정도에 대해서

- 라벨은 공통된 내용이 여러 곳에서 사용될 것으로 예상되므로, as prop에 미리 정의된 키워드를 입력하면 styled 내부에서 알아서 분기시켜주는 방향으로 구현.
- bgColor와 children도 받을 수 있으나, 둘 중 하나만 입력하는 경우에는 "텍스트와 bgColor를 동시에 입력해주세요" 문구를 렌더링함으로써 사실상 사용 불가능하도록 만듬

wherehows committed:
Feat: StatusTag 마크업 추가

wherehows committed:
Chore: Conflict 해결

sanoopark commented on src/components/StatusTag/StatusTag.jsx:
```diff
@@ -0,0 +1,49 @@
+
+StatusTag.defaultProps = {
+  children: '상태 없음',
+  padding: '0.6rem 0.6rem',
```
```suggestion
  padding: '0.6rem',
```

sanoopark commented on src/components/StatusTag/StatusTag.jsx:
```diff
@@ -0,0 +1,49 @@
+  backgroundColor: '#ADADAD',
+  width: '5.6rem',
+  fontSize: '1.6rem',
+  borderRadius: '0 1.6rem 0 1.6rem'
```
```suggestion
  borderRadius: '0 1.6rem'
```

wherehows committed:
Refactor: 코드 리뷰 반영

- defaultProps 대신 단축 평가를 이용한 defaultValue 할당
- as에 따라서 특정한 라벨 사용 가능
- CSS 컨벤션 준수

sanoopark commented on src/components/StatusTag/StatusTag.jsx:
```diff
@@ -0,0 +1,73 @@
+  return theme.colors.lighterGray;
+};
+
+const decideText = (as, bgColor, children) => {
+  if (bgColor && children) {
+    return children;
+  }
+
+  if ((bgColor && !children) || (!bgColor && children)) return '배경색과 텍스트 둘 다 입력해주세요';
+
+  if (as) {
+    return STATUS[as];
+  }
+
+  return '내용을 입력해 주세요';
+};
```
children으로 들어가는 것이 사용자에게 보여주는 메시지인가요? 개발자가 보는 메시지인가요?
아래 같이 써도 동작은 하는데 어떤 부분을 염두에 두신 건지 궁금합니다

```
      <StyledDiv>
         {children || (as && decideText(as))}
      </StyledDiv>
```

```suggestion
const decideText = (as) => {
  return STATUS[as];
};
```

wherehows replied:
as, bgColor, children을 입력받을 수 있는데, 그렇게 되면 총 경우의 수가 8가지가 됩니다.

이중 bgColor와 children중 적어도 하나를 사용했다는 것은 label을 커스톰하겠다는 의미로 처리하기 위해, 둘 중 하나만 입력하는 경우, 화면에 엉뚱한 텍스트를 출력하여 개발자에게 알려주기 위함입니다.

sanoopark commented on src/components/StatusTag/StatusTag.jsx:
```diff
@@ -0,0 +1,73 @@
+const Wrapper = styled.div``;
+
+const StyledDiv = styled.div`
+  background-color: ${(props) => decideColor(props)};
```
StyledDiv에 as prop이 안 들어가서 decideColor에서 as가 undefined로 찍히고 색상도 안 나오는 것 같은데 확인 부탁드립니다

wherehows committed:
Refactor: StatusTag as prop이름 status로 수정

wherehows committed:
Refactor: StatusTag의 StyledDiv를 StyledSpan로 변경

- height 및 width prop 삭제

wherehows committed:
Chore: Conflict 해결

wherehows committed:
Chore: Conflict 해결

wherehows committed:
Refactor: 코드 컨벤션 준수

wherehows committed:
Refactor: constants 폴더에 STATUS 네임 스페이스 추가

ghost commented:
| index | status | bgColor | children | Y/N | Error |
|-------|--------|---------|----------|-----|-------|
|   1   |    o   |    x    |     x    |  Y  |   x   |
|   2   |    x   |    o    |     o    |  Y  |   x   |
|   3   |    o   |    o    |     x    |  N  |   o   |
|   4   |    o   |    x    |     o    |  N  |   o   |
|   5   |    x   |    o    |     x    |  N  |   o   |
|   6   |    x   |    x    |     o    |  N  |   o   |
|   7   |    x   |    x    |     x    |  N  |   o   |
|   8   |    o   |    o    |     o    |  N  |   x   |

제가 8가지 경우를 테스트해본 결과인데요. 8번은 저희 얘기했던 대로라면 에러를 줘야 하는 경우인 것 같은데 확인 부탁드립니다! 그리고 isWronglyUsed에서 추상화된 함수들이 단축 평가보다 가독성이 조금 떨어지는 것 같은데 여기에 대해서도 이야기해보고 싶습니다!

wherehows committed:
Refactor: isWronglyUsed 조건 분기문 함수 코드 수정

wherehows committed:
Chore: Conflict 해결
